data/src/main/java/me/androidbox/data/alarm_manager/AlarmReceiver.kt

```diff
@@ -1,15 +1,69 @@
 package me.androidbox.data.alarm_manager
 
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.PendingIntent
 import android.content.BroadcastReceiver
 import android.content.Context
 import android.content.Intent
+import android.os.Build
+import androidx.core.app.NotificationCompat
+import me.androidbox.data.R
+import me.androidbox.data.alarm_manager.AlarmSchedulerImp.Companion.EXTRA_AGENDA_TYPE
+import me.androidbox.data.alarm_manager.AlarmSchedulerImp.Companion.EXTRA_DESCRIPTION
+import me.androidbox.data.alarm_manager.AlarmSchedulerImp.Companion.EXTRA_ID
+import me.androidbox.data.alarm_manager.AlarmSchedulerImp.Companion.EXTRA_TITLE
 
 class AlarmReceiver : BroadcastReceiver() {
 
     override fun onReceive(context: Context?, intent: Intent?) {
-        intent?.getStringExtra("EXTRA_MESSAGE")?.let { message ->
-            /* TODO Add Notification here */
-            println("Alarm triggered: $message")
+        /* TODO Not sure if intent and context can be null but I guess is ok to cast it anyway */
+        if(intent != null && context != null) {
+            intent.extras?.let { bundle ->
+                val agendaId = bundle.getString(EXTRA_ID) ?: ""
+                val title = bundle.getString(EXTRA_TITLE) ?: "Title"
+                val description = bundle.getString(EXTRA_DESCRIPTION) ?: "Description"
+                val agendaType = bundle.getString(EXTRA_AGENDA_TYPE) ?: "Agenda"
+
+                if (agendaId.isNotBlank()) {
+                    createNotificationChannel(context, agendaType)
```

Comment:
I'd create the notification channel in `onCreate` of your application class, so directly when your app launches. Channels are used so that users can decide in the app settings whether they want to receive a certain type of notifications or not. If you create the channel just before showing a notification, they won't be able to change that setting before